Sources/ProtobufferCoding/Encoder/ProtoEncodingContainer.swift

```diff
@@ -108,6 +108,16 @@ internal class InternalProtoEncodingContainer {
     }
 
     internal func encodeInt(_ value: Int, tag: Int) throws {
+        if let strategy = encoder.variableWidthIntegerStrategy {
+            switch strategy {
+            case .thirtyTwo:
+                try encodeInt32(Int32(value), tag: tag)
+            case .sixtyFour:
+                try encodeInt64(Int64(value), tag: tag)
+            }
+            return
+        }
+        
         if MemoryLayout<Int>.size == 4 {
             try encodeInt32(Int32(value), tag: tag)
         } else if MemoryLayout<Int>.size == 8 {
```

Comment:
Maybe instead of `encoder.variableWidthIntegerStrategy` being optional and keeping those `MemoryLayout.size` checks, we could make `encoder.variableWidthIntegerStrategy` non-optional and set it to the system's architecture-width by default in the constructor of the encoder?

Reply:
I do that now: 40da4b12a554d66823168ba28484d7f3c668ce7f